### PR TITLE
Hopefully completely fix printf-family %a rounding

### PR DESCRIPTION
**Pull request by GabrielRavier**
The a conversion specifier to printf had some issues w.r.t. rounding, in
particular in edge cases w.r.t. "to nearest, ties to even" rounding (for
instance, "%.1a" with 0x1.78p+4 outputted 0x1.7p+4 instead of 0x1.8p+4).

This patch fixes this and adds several tests w.r.t ties to even rounding

### CODE REVIEW

File: libc/stdio/fmt.c [hunk 1]
```diff
@@ -701,22 +704,23 @@ static int __fmt_bround(struct FPBits *b, int prec, int prec1) {
...
   // doesn't need to be set as we have already done so above)
-  // The last one handles rounding to nearest
-  current_rounding_mode = fegetround();
-  if (current_rounding_mode == FE_TOWARDZERO ||
-      (current_rounding_mode == FE_UPWARD && b->sign) ||
-      (current_rounding_mode == FE_DOWNWARD && !b->sign))
+  // They use the FLT_ROUNDS value - i.e. 0 -> toward zero, 1 -> to nearest,
+  // ties to even, 2 -> toward positive infinity, 3 -> towards negative infinity
+  // The last if handles rounding to nearest
+  if (b->fpi.rounding == 0 || (b->fpi.rounding == 2 && b->sign) ||
```
Comment (jart) [inline libc/stdio/fmt.c:710]: Wouldn't it be better to say `FPI_Round_zero` and `FPI_Round_up` here?

**Review by jart — CHANGES_REQUESTED** [review]
This is impressive. Could you avoid the magic numbers?

File: libc/stdio/fmt.c (hunk 1 above)
Comment (GabrielRavier) [reply libc/stdio/fmt.c]: This commit now uses the `FPI_Round_*` values.

**Comment by GabrielRavier** [comment]
The commit now uses the `FPI_Round_*` enum values as appropriate.

**Review by jart — APPROVED** [review]
